crates/fluvio-spu/src/replication/leader/replica_state.rs

```diff
@@ -384,6 +391,20 @@ where
     pub async fn followers_info(&self) -> BTreeMap<SpuId, OffsetInfo> {
         self.followers.read().await.clone()
     }
+
+    pub async fn register_offset_publisher(&self, offset_publisher: &Arc<OffsetPublisher>) {
+        self.consumer_offset_publishers
+            .lock()
+            .await
+            .push(offset_publisher.clone());
+    }
+
+    pub async fn signal_topic_deleted(&self) {
+        let offset_publishers = self.consumer_offset_publishers.lock().await;
+        for publisher in offset_publishers.iter() {
+            publisher.update(TOPIC_DELETED);
+        }
+    }
 }
```

Comment:
would need some periodic clean up to ensure stale publishers are removed.

Reply:
Downgraded the pointers to be weak pointers (to allow dropping of values) and changed the registration logic to check/filter the list for dead pointers every ~10 registrations. Ideally this means that you're limited to, at most, 10 wasted allocations. Maybe this isn't so great if we expect extreme numbers of consumers since this is just a simple vector, but still O(n) * (time it takes to upgrade/downgrade pointers).